pkg/assets/handlers.go

```diff
@@ -154,6 +154,22 @@ type WebConsoleVersion struct {
 	OpenShiftVersion  string
 }
 
+var extensionPropertiesTemplate = template.Must(template.New("webConsoleExtensionProperties").Parse(`
+window.OPENSHIFT_EXTENSION_PROPERTIES = {
+  {{ $values := .ExtensionPropertyValues }}
+  {{ range $i, $key := .ExtensionPropertyKeys }}
+  {{ if $i }},{{ end }}
+  {{ $value := index $values $i }}
+  "{{ $key | js }}": "{{ $value | js }}"
```

Comment:
Yep js is doing the escaping
On May 19, 2016 8:30 AM, "David Eads" <notifications@github.com> wrote:

> In pkg/assets/handlers.go
> <https://github.com/openshift/origin/pull/8934#discussion_r63868984>:
>
> > @@ -154,6 +154,22 @@ type WebConsoleVersion struct {
> >  	OpenShiftVersion  string
> >  }
> >
> > +var extensionPropertiesTemplate = template.Must(template.New("webConsoleExtensionProperties").Parse(`
> > +window.OPENSHIFT_EXTENSION_PROPERTIES = {
> > +  {{ $values := .ExtensionPropertyValues }}
> > +  {{ range $i, $key := .ExtensionPropertyKeys }}
> > +  {{ if $i }},{{ end }}
> > +  {{ $value := index $values $i }}
> > +  "{{ $key | js }}": "{{ $value | js }}"
>
> This doesn't look like its doing escaping, but the doc here
> https://github.com/openshift/origin/pull/8934/files#diff-3788994f990386b63387ece5c3b5dc33R570
> doesn't mention any need for the user to escape values himself.
>
> Oh, that js function that does it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8934/files/dd6185b4128407493039a8ab0e055ba3d217c370#r63868984>
>